src/container/MatchingQuestions/useMatchingQuestions.ts

```diff
@@ -0,0 +1,325 @@
+import { FormEvent, useState, useEffect } from 'react'
+import {
+  EMatchingQuestionsSteps,
+  IMatchingQuestionsForm,
+  IQuestion,
+  IQuestionChoice,
+} from '@/types/pages/matchingQuestions'
+import { array, string, number, object, boolean } from 'yup'
+import { FieldErrors, useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import useRequestHandlers from '@/hooks/useRequestHandlers'
+import useHandleError from '@/hooks/useHandleError'
+import { IFrameWork } from '@/types/pages/finishSignup'
+
+const schema = object({
+  basicProgrammingLanguagesKnowledge: array().min(1).required(),
+  proficientProgrammingLanguages: array().min(1).required(),
+  preferredLearningStyle: array().min(1).required(),
+  learningFrequency: array().min(1).required(),
+  preferredCommunicationMethod: array().min(1).required(),
+  technologyOfInterest: array().min(1).required(),
+  weeklyHoursDedicatedToLearningAndCollaboration: number().required(),
+  motivationForLearningAndCollaboration: array().min(1).required(),
+  TrackOfInterest: array().min(1).required(),
+  goalsOnThePlatform: array().min(1).required(),
+  comfortLevelWithRemoteWorkOrCollaboration: object().required(),
+  projectTypeInterest: array().min(1).required(),
+})
+const useMatchingQuestions = () => {
+  const [currentStep, setCurrentStep] = useState<EMatchingQuestionsSteps>(0)
+
+  const [allFieldsData, setAllFieldsData] = useState<IQuestionChoice[]>([])
+
+  const [frameWorks, setFrameWorks] = useState<IFrameWork[]>([])
+
+  const { postHandler, loading, getHandler } = useRequestHandlers()
+
+  const { handleError } = useHandleError()
+  const getData = async (): Promise<void> => {
+    const { data, error } = await getHandler({ endpoint: '/submissions/phaseone' })
+    if (error) return handleError(error)
+    setAllFieldsData(data)
+  }
+
+  const defaultValues = {
+    basicProgrammingLanguagesKnowledge: [],
+    proficientProgrammingLanguages: [],
+    preferredLearningStyle: [],
+    learningFrequency: [],
+    preferredCommunicationMethod: [],
+    TrackOfInterest: [],
+    technologyOfInterest: [],
+    weeklyHoursDedicatedToLearningAndCollaboration: '',
+    motivationForLearningAndCollaboration: [],
+    goalsOnThePlatform: [],
+    comfortLevelWithRemoteWorkOrCollaboration: null,
+    projectTypeInterest: [],
+  }
+
+  const {
+    formState: { errors },
+    control,
+    handleSubmit,
+    watch,
+  } = useForm<IMatchingQuestionsForm>({
+    resolver: yupResolver(schema),
+    defaultValues,
+  })
+
+  const tracksValue = watch('TrackOfInterest')
+
+  useEffect(() => {
+    const frameWorks = tracksValue.flatMap((track, frameworks) => track.frameworks)
+    setFrameWorks(frameWorks)
+  }, [tracksValue])
+  useEffect(() => {
+    ;(async (): Promise<void> => {
```

Comment:
you don't need this function you can just call getData function directly on the useEffect because you don't need to wrap it in try catch, getData function is already handling error it self